### DISCUSSION TITLE
MariaDB@10.11.8 Bottle Service - Unsafe Host Shutdown Corrupts InnoDB?

### DISCUSSION THREAD

**Discussion by troyxmccall** [Homebrew/discussions#5389] [Everyday usage] [answered]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.3.1-13-gb9a9b0a
ORIGIN: https://github.com/Homebrew/brew
HEAD: b9a9b0a5d10e3dda8148fde434456441c4db0f52
Last commit: 14 hours ago
Core tap HEAD: ba592bd9f41b9edf3e8f69d119be11447354c45c
Core tap last commit: 18 minutes ago
Core tap JSON: 22 May 13:44 UTC
Core cask tap HEAD: 6e0f78e8b8fe1f46e02d0daf02892d5765d6860d
Core cask tap last commit: 28 minutes ago
Core cask tap JSON: 22 May 13:44 UTC
HOMEBREW_PREFIX: /usr/local
HOMEBREW_BAT_THEME: Dracula
HOMEBREW_CASK_OPTS: ["--appdir=/Applications"]
HOMEBREW_DISPLAY: /private/tmp/com.apple.launchd.Zk2v8NYDGN/org.xquartz:0
HOMEBREW_EDITOR: vim
HOMEBREW_MAKE_JOBS: 12
HOMEBREW_NO_ANALYTICS: set
HOMEBREW_SORBET_RUNTIME: set
Homebrew Ruby: 3.3.1 => /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/3.3.1/bin/ruby
CPU: dodeca-core 64-bit westmere
Clang: 14.0.0 build 1400
Git: 2.45.1 => /usr/local/bin/git
Curl: 8.4.0 => /usr/bin/curl
macOS: 12.7.5-x86_64
CLT: 14.2.0.0.1.1668646533
Xcode: 14.2
```

### Output of `brew doctor`

```shell
N/A
```

### Description of issue

there seems to be an issue when running the latest mariadb@10.11.8 bottle as a brew service

shutting down your host while the brew service is running (even if you don't have any active Queries/Procedures) seems to corrupt InnoDB

on a macOS reboot, the brew service will fail with 

```
tail /usr/local/var/mysql/my-mac.err
```

```
2024-05-21 11:44:56 0 [Note] InnoDB: Completed initialization of buffer pool
2024-05-21 11:44:56 0 [ERROR] InnoDB: Missing FILE_CHECKPOINT(102890851) at 110803559
2024-05-21 11:44:56 0 [ERROR] InnoDB: Log scan aborted at LSN 110803559
2024-05-21 11:44:56 0 [ERROR] InnoDB: Plugin initialization aborted with error Generic error
2024-05-21 11:44:56 0 [Note] InnoDB: Starting shutdown...
2024-05-21 11:44:56 0 [ERROR] Plugin 'InnoDB' registration as a STORAGE ENGINE failed.
2024-05-21 11:44:56 0 [Note] Plugin 'FEEDBACK' is disabled.
2024-05-21 11:44:56 0 [ERROR] Unknown/unsupported storage engine: InnoDB
2024-05-21 11:44:56 0 [ERROR] Aborting
240521 11:44:56 mysqld_safe mysqld from pid file /usr/local/var/mysql/my-mac.pid ended
```

there are ways around this, like setting 

```
[mariadb]
innodb_force_recovery = 6
``` 

in `/usr/local/etc/my.cnf.d/my.cnf`

but then you have to 
- repair all your tables
-  export all your tables to a sql file, 
- delete all your tables (redundant if you also delete the datadir below, but let's do it all) 
- shutdown mariadb, 
- remove your data dir `/usr/local/var/mysql/` just to be sure
- also clean out your `/usr/local/etc/my.cnf.d/*` folder, again just to be sure with a fresh install (also backup then remove your ` ~/.my.cnf` 
- start mariadb
- re-import

and if you restart macOS again, with brew services running you'll brick `InnoDB` all over again and have to do this repair process again

**Comment by tobycurl** [comment]
I have had similar corruption on that version of MariaDB.

After getting a backup of all the databases I emptied out my whole installation to try to get it working again.

I found that stopping the service with "brew services stop mariadb@10.11" claimed it had stopped but the process was still running and at 100% CPU. This meant that MariaDB couldn't then be restarted.

I also tried version 10.6 which had the same issue.

I have now installed the latest version of MariaDB (11.3.2). This does seem to stop the process correctly when stopping the service.

So it's maybe an issue with the current 10.X versions?

Output of brew config:

```
HOMEBREW_VERSION: 4.3.1
ORIGIN: https://github.com/Homebrew/brew.git
HEAD: e0bc557e7b991cb23583679e1cf1c8a92b793aeb
Last commit: 4 days ago
Core tap JSON: 24 May 08:10 UTC
Core cask tap JSON: 24 May 08:10 UTC
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_EDITOR: subl
HOMEBREW_MAKE_JOBS: 6
Homebrew Ruby: 3.3.1 => /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/3.3.1/bin/ruby
CPU: hexa-core 64-bit kabylake
Clang: 15.0.0 build 1500
Git: 2.45.1 => /usr/local/bin/git
Curl: 8.4.0 => /usr/bin/curl
macOS: 14.4.1-x86_64
CLT: 15.3.0.0.1.1708646388
Xcode: 15.4
```

**Reply by zsavajji** [reply]
Got the same issue on Sonoma 14.5 and MariaDB 11.4.2 

```
2024-06-05  9:00:39 0 [Note] Starting MariaDB 11.4.2-MariaDB source revision 3fca5ed772fb75e3e57c507edef2985f8eba5b12 as process 3374
2024-06-05  9:00:39 0 [Warning] Setting lower_case_table_names=2 because file system for /opt/homebrew/var/mysql/ is case insensitive
2024-06-05  9:00:39 0 [Note] InnoDB: Compressed tables use zlib 1.2.12
2024-06-05  9:00:39 0 [Note] InnoDB: Number of transaction pools: 1
2024-06-05  9:00:39 0 [Note] InnoDB: Using generic crc32 instructions
2024-06-05  9:00:39 0 [Note] InnoDB: Initializing buffer pool, total size = 128.000MiB, chunk size = 2.000MiB
2024-06-05  9:00:39 0 [Note] InnoDB: Completed initialization of buffer pool
2024-06-05  9:00:39 0 [Note] InnoDB: Resetting space id's in the doublewrite buffer
2024-06-05  9:00:39 0 [ERROR] InnoDB: Missing FILE_CHECKPOINT(84933786464) at 84933786464
2024-06-05  9:00:39 0 [ERROR] InnoDB: Log scan aborted at LSN 84933786464
2024-06-05  9:00:39 0 [ERROR] InnoDB: Plugin initialization aborted with error Generic error
2024-06-05  9:00:39 0 [Note] InnoDB: Starting shutdown...
2024-06-05  9:00:39 0 [ERROR] Plugin 'InnoDB' registration as a STORAGE ENGINE failed.
2024-06-05  9:00:39 0 [Note] Plugin 'FEEDBACK' is disabled.
2024-06-05  9:00:39 0 [Note] Plugin 'wsrep-provider' is disabled.
2024-06-05  9:00:39 0 [Warning] 'innodb-file-format' was removed. It does nothing now and exists only for compatibility with old my.cnf files.
2024-06-05  9:00:39 0 [ERROR] Unknown/unsupported storage engine: InnoDB
2024-06-05  9:00:39 0 [ERROR] Aborting
```

I resolved the issue by truncating the `ib_logfile0`, that made mariaDB restart without errors. 

Could it be related to brew or to the macOS update? 

Thanks

**Reply by stychos** [reply]
> Got the same issue on Sonoma 14.5 and MariaDB 11.4.2
> 
> ```
> 2024-06-05  9:00:39 0 [Note] Starting MariaDB 11.4.2-MariaDB source revision 3fca5ed772fb75e3e57c507edef2985f8eba5b12 as process 3374
> 2024-06-05  9:00:39 0 [Warning] Setting lower_case_table_names=2 because file system for /opt/homebrew/var/mysql/ is case insensitive
> 2024-06-05  9:00:39 0 [Note] InnoDB: Compressed tables use zlib 1.2.12
> 2024-06-05  9:00:39 0 [Note] InnoDB: Number of transaction pools: 1
> 2024-06-05  9:00:39 0 [Note] InnoDB: Using generic crc32 instructions
> 2024-06-05  9:00:39 0 [Note] InnoDB: Initializing buffer pool, total size = 128.000MiB, chunk size = 2.000MiB
> 2024-06-05  9:00:39 0 [Note] InnoDB: Completed initialization of buffer pool
> 2024-06-05  9:00:39 0 [Note] InnoDB: Resetting space id's in the doublewrite buffer
> 2024-06-05  9:00:39 0 [ERROR] InnoDB: Missing FILE_CHECKPOINT(84933786464) at 84933786464
> 2024-06-05  9:00:39 0 [ERROR] InnoDB: Log scan aborted at LSN 84933786464
> 2024-06-05  9:00:39 0 [ERROR] InnoDB: Plugin initialization aborted with error Generic error
> 2024-06-05  9:00:39 0 [Note] InnoDB: Starting shutdown...
> 2024-06-05  9:00:39 0 [ERROR] Plugin 'InnoDB' registration as a STORAGE ENGINE failed.
> 2024-06-05  9:00:39 0 [Note] Plugin 'FEEDBACK' is disabled.
> 2024-06-05  9:00:39 0 [Note] Plugin 'wsrep-provider' is disabled.
> 2024-06-05  9:00:39 0 [Warning] 'innodb-file-format' was removed. It does nothing now and exists only for compatibility with old my.cnf files.
> 2024-06-05  9:00:39 0 [ERROR] Unknown/unsupported storage engine: InnoDB
> 2024-06-05  9:00:39 0 [ERROR] Aborting
> ```
> 
> I resolved the issue by truncating the `ib_logfile0`, that made mariaDB restart without errors.
> 
> Could it be related to brew or to the macOS update?
> 
> Thanks

Same on Ventura, looks like not a macOS update issue.

**Comment by dotsam** [comment, marked as answer]
It looks like this is an upstream bug ([MDEV-34129](https://jira.mariadb.org/browse/MDEV-34129) and [MDEV-34254](https://jira.mariadb.org/browse/MDEV-34254)) and has a possible pending fix for the 10.11.x series in MariaDB/server#3253

**Comment by camille-hdl** [comment]
I have this problem as well.   

With 10.11, mariadbd fails to stop with the message "Signal handler thread did not exit in a timely manner. Continuing to wait for it to stop..".   
With 11.4 however, mariadbd shuts down correctly (at least, the process ends), but the "Missing FILE_CHECKPOINT" InnoDB error is still occurring.   
Truncating `ib_logfile0` does not allow MariaDB to restart, only giving the message `[ERROR] InnoDB: ib_logfile0 is empty, and LSN is unknown.` since, as I understand it, a valid log file is required by InnoDB now.

The only way to recover i've found is to completely remove mariadb, reinstall and re-import the databases.

I have tried using `brew services`, `mysql.server start/stop`, `mysqld_safe` or `mysqld` directly, without success.

I'm using macOS 14.5 on Apple silicon.

**Comment by camille-hdl** [comment]
in case that's relevant, the issue does not happen with mysql 5.7 on the same system with the same database.

**Comment by mazzelman** [comment]
I updated my machine yesterday, and i think i have the same problem. I uninstalled mariabd and installed it again. I disabled the auto start. But it still crashes. This is really annoying. As i am not that good with mariadb, is there a solution for the moment? Can i go back to a previous version of mariadb?

**Comment by mazzelman** [comment]
I have now installed an older version. That is working for now. I will describe what i did. But please keep in mind, that i have this running here on my mac for testing purposes. I'm not responsible when something breaks! This will erase all of your database files. So this is only working, if you have a backup!

1. uninstall mariadb

`brew uninstall mariadb` 

Brew says that the mecab files are still there, so i did a:

`sudo rm -rf /opt/homebrew/etc/mecabrc`

After that i removed other files too:

`rm -rf /opt/homebrew/var/mysql`
`rm -rf /opt/homebrew/etc/my.cnf`
`rm -rf /opt/homebrew/etc/my.cnf.d`
`rm -rf /opt/homebrew/etc/my.cnf.default`

Then i restarted the mac.

2. install mariadb@10.10

To install mariadb i choosed an older version.

`brew install mariadb@10.10`

After the install brew mentions that this version is not in the path, because it is an older version. But brew tells me, that when i need this in my path (to work with it in the console) i can do a:

`echo 'export PATH="/opt/homebrew/opt/mariadb@10.10/bin:$PATH"' >> ~/.zshrc`

Important! I use zsh. If you use another terminal like bash, i hope brew will give you the right copy and paste command!

Restart the terminal so that you can use the command mysql

3. Set a root password

`sudo mysql`

here you have to type in your account password, that you use for your mac user!

If you typed in your password correct, you will enter this mysql command thingy...

Here i did:

`ALTER USER 'root'@'localhost' IDENTIFIED BY 'newrootpassword';`

Important! Replace newrootpassword with your own password. You can choose a new pasword for the mysql root user.

Then write exit to leave the mysql console

`exit`

After that i imported back my databases with phpmyadmin. And did a whole restart of the system. And it works for me again!

I hope this helps for now. 🖖

**Reply by joubertvasc** [reply]
Worked for me. Thank you very much. The version 11.4.2 is broken.

**Reply by abubakar-iqbal** [reply]
Thanks you soo much it's worked for me as well. But you don't mention the mysql data will be loss.

**Reply by suleodu** [reply]
I got ERROR 2002 (HY000): Can't connect to local server through socket '/tmp/mysql.sock' (2) when i ran sudo mysql

**Reply by ghost** [reply]
you need to start the service with `brew services start mariadb@10.10`

**Reply by 1rocketdude** [reply]
I confirm that version 11.4.2 (latest) is similarly broken. I manually tried 11.2.4, and this was broken as well. Do we really roll back all the way back to 10.10?

**Reply by MarH0L9** [reply]
Thanks for the tip. I updated brew and got the lastest version of mariadb, 11.4.2 and always breaks after shuting down and starting my macbook pro.

I followed your steps, but I installed, mariadb 11.1 with brew. Followed all the other steps (also used the command in terminal and it worked for me).

Now everything is working well. Hopefully I have backup of all my db.

*Remember to make backups guys!

Edit: nah, 11.1 is broken also. Yesterday I installed it, shut down my mac, then started it, and everything was fine. But this morning when I started my mac, same problem again.

Today I installed 10.6.18, I hope this time everything works properly 🤞🏻 (at least seems like everything is good now).

**Comment by apexdivision** [comment]
I am having the same problem with MariaDB latest 11.4.2 on macOS 14.5 on M1.
Going ahead as suggested by @mazzelman for now.
Ironically could not find much info on this version error on net, is it only me who is having this issue?

**Reply by joubertvasc** [reply]
No, I was blowing my mind for days until find this post.

**Reply by 1rocketdude** [reply]
same here. No luck until this post. Do we really roll back all the way back to 10.10?

**Comment by mikeho** [comment]
Having this issue as well... going to downgrade to an older version, similar to above...

But @mazzelman just a FYI -- 10.10 is deprecated and not considered LTS.

See [MariaDB Server Releases](https://mariadb.com/kb/en/mariadb-server-release-dates/) for more info.

I've gone ahead and downgraded to 10.6 (LTS).

**Comment by mazzelman** [comment]
@mikeho 
Hi, thanks for pointing out. 👍

**Comment by renekreijveld** [comment]
I have the same problems in MariaDB 11.4.2 and 11.2.4.
I initially downgraded to MariaDB 11.2.4, but that had the same problem. After uninstalling everything I installed 10.6.18.
With that version I experience the following:

After starting MariaDB with:

`brew services start mariadb@10.6`

I see the following processes:

`  501 13392     1   0  2:52pm ??         0:00.02 /bin/sh /opt/homebrew/opt/mariadb@10.6/bin/mysqld_safe --datadir=/opt/homebrew/var/mysql`
`  501 13505 13392   0  2:52pm ??         0:00.62 /opt/homebrew/opt/mariadb@10.6/bin/mariadbd --basedir=/opt/homebrew/opt/mariadb@10.6 --datadir=/opt/homebrew/var/mysql --plugin-dir=/opt/homebrew/opt/mariadb@10.6/lib/plugin --log-error=/opt/homebrew/var/mysql/MacBook-Pro.local.err --pid-file=MacBook-Pro.local.pid`

Then, when I stop MariaDB with the following:

`brew services stop mariadb@10.6`

I get a conformation that MariaDB is stopped:

`==> Successfully stopped 'mariadb@10.6' (label: homebrew.mxcl.mariadb@10.6)`

But if I check the running processes I still see these:

`  501 13505     1   0  2:52pm ??         0:52.26 /opt/homebrew/opt/mariadb@10.6/bin/mariadbd --basedir=/opt/homebrew/opt/mariadb@10.6 --datadir=/opt/homebrew/var/mysql --plugin-dir=/opt/homebrew/opt/mariadb@10.6/lib/plugin --log-error=/opt/homebrew/var/mysql/MacBook-Pro.local.err --pid-file=MacBook-Pro.local.pid`

I can kill that process with kill -9. If I then restart MariaDB there is no InnoDB corruption. So this way I have a usable workaround.

**Reply by reckart** [reply]
I have the impression that issue started with Sonoma... I only ran into it after upgrading the OS yesterday.

I wonder if anybody can confirm the issue with Ventura.

**Comment by landwire** [comment]
I have the same problem on an M3 Apple with 11.4.2.
Any fix in the pipeline?

**Reply by renekreijveld** [reply]
I downgraded to version 10.6.18. That solved it for now.

**Reply by joubertvasc** [reply]
I downgraded to 10.10 and my head is on my neck again.

**Reply by troyxmccall** [reply]
I just moved to run MariaDB on docker since versioning is critical in my dev workflow. 

```yml
services:

  mariadb:
    image: mariadb:10.11
    restart: unless-stopped
    command: --default-authentication-plugin=mysql_native_password
    environment:
      MYSQL_ROOT_PASSWORD: yourNewRootPasswordHere
      MYSQL_DATABASE: mydatabase
      MYSQL_USER: docker
      MYSQL_PASSWORD: yourNewDockerPasswordHere
    volumes:
     - mariadb_10_11:/var/lib/mysql
    ports:
      - "3306:3306"

  automysqlbackup:
    image: databack/mysql-backup
    restart: unless-stopped
    environment:
      DB_SERVER: mariadb
      DB_TYPE: mysql
      DB_HOST: mariadb
      DB_PORT: 3306
      DB_USER: root
      DB_PASS: yourNewRootPasswordHere
      DB_DUMP_FREQ: 1440
      DB_DUMP_TARGET: /backup
    volumes:
      - automysqlbackup:/backup
      - mariadb_10_11:/var/lib/mysql:ro # Read-only access to MariaDB data

volumes:
  mariadb_10_11:
  automysqlbackup:      
```

**Comment by reckart** [comment]
I upgraded to Sonoma (Apple M2) yesterday and that is where the problems started.

FWIW, I have tried 11.4, 10.10 and 10.6 now and all seem to have the problem that `brew services stop mariadb...` leaves the mariadb server process hanging around.

I get a message about the showdown in the log:

```
2024-06-20  8:16:53 0 [Note] /opt/homebrew/opt/mariadb@10.6/bin/mariadbd (initiated by: unknown): Normal shutdown
```

but running `ps aux | grep mysql`, I can see that the process continues to be there

```
myuser         40589   0.0  0.3 411358560  73552   ??  S     8:16AM   1:38.39 /opt/homebrew/opt/mariadb@10.6/bin/mariadbd --basedir=/opt/homebrew/opt/mariadb@10.6 --datadir=/opt/homebrew/var/mysql --plugin-dir=/opt/homebrew/opt/mariadb@10.6/lib/plugin --log-error=/opt/homebrew/var/mysql/myhost.err --pid-file=myhost.pid
```

Even `kill PID` does not work. When I try that, I get another message in the log 

```
2024-06-20  8:18:33 0 [Warning] Signal handler thread did not exit in a timely manner. Continuing to wait for it to stop...
```

And I think that is the root cause of the problem: the server process that is not stopping. 

Eventually, when the system is restarted, it will be force-stopped - which is bad for the data files. Or when trying to stop/restart the service (in particular with 11.4), it seems that the service script might not be checking properly if another deamon is already running which can then lead to a corrupted redo log.

**Reply by reckart** [reply]
With Mariadb 11.4.2, the service seems to stop properly - but when I restart the service I have the corrupt DB.

**Reply by legentbv** [reply]
When I downgraded to 10.6.18 stopping MariaDB also doesn't work. I need to kill the process with `kill -9`.
But, the databases don't get corrupted. If I don't kill the process and reboot the Macbook the databases also don't get corrupted.

**Reply by reckart** [reply]
* https://jira.mariadb.org/browse/MDEV-34350 is about the broken ib_logfile0 after restart - no fix yet
* https://jira.mariadb.org/browse/MDEV-34129 is about the process not shutting down - this should be fixed in 10.6.19 when it is released

**Comment by stychos** [comment]
I tried all LTS versions available in Homebrew, all are broken. My databases are big and always get corrupted on forced shutdown. Switched to Percona Server for now, but it is not available on ARM.

Upd: I'm on Ventura.

**Comment by klodeckl** [comment]
Same here. I downgraded to 10.6. But process stays up when stopping as described above, kill -9 process pid works then. A soon fix would be very helpful.

**Comment by magic-thomas** [comment]


**Reply by reckart** [reply]
The issue is already being worked on: https://jira.mariadb.org/browse/MDEV-34422

From the comments, downgrading to anything before 10.8 should help. However, until a 10.6.19 you will then hit https://jira.mariadb.org/browse/MDEV-34129.

**Comment by sokolnikov911** [comment]
I have the same problem after upgrading to the latest mariadb version. So I decided to downgrade to version **10.6.18**, and it is working well for me.

**Comment by 1rocketdude** [comment]
as I have tried downgrading from 11.4.2 to 11.2.4 with the same result - corrupted InnoDB files that prevent startup.
InnoDB: Missing FILE_CHECKPOINT
[ERROR] Plugin 'InnoDB' registration as a STORAGE ENGINE failed.

on both versions 11.4.2 to 11.2.4, even a manual 'mariadb-admin flush-tables' before shutdown DOES NOT prevent the table corruption.
'mariadb-admin shutdown' takes a very long time (>30 seconds) to actually shut the server down.

**Comment by drewcking** [comment]
The Jira issue (https://jira.mariadb.org/browse/MDEV-34422) is showing closed as Fixed as of this morning. It looks like it'll be part of the 11.4.3 release, but I don't see an indication when that'll be available. MariaDB's patch release cadence is about quarterly, and 11.4.2 came out on May 29th, so maybe end of August?

**Comment by nirjhareswar** [comment]
No indication in Jira, which rolled out version will have this fixed.

**Comment by renekreijveld** [comment]
https://jira.mariadb.org/browse/MDEV-34422 indicates:
Resolution: fixed
Fix Version/s: 10.11.9, 11.1.6, 11.2.5, 11.4.3, 11.5.2, 11.6.1

**Reply by nirjhareswar** [reply]
My bad, missed it. Thank you for pointing.

**Reply by nirjhareswar** [reply]
These versions are not yet released for homebrew.

[https://formulae.brew.sh/formula/mariadb](url)

**Reply by renekreijveld** [reply]
Yes, see this comment: https://github.com/orgs/Homebrew/discussions/5389#discussioncomment-10193932

**Comment by matthewtoschi** [comment]
Looks like I can't downgrade anymore either. Hope the fixed version is released soon.

`mariadb@10.10 has been disabled because it is not supported upstream! It will be disabled on 2024-07-26.`

**Reply by renekreijveld** [reply]
You can downgrade to MariaDB 10.6.18.

**Reply by vorsprungRene** [reply]
You can still downgrade to mariadb@10.10 via:

Tap homebrew-core:

`brew tap homebrew/core --force`


Edit the formula with:

`brew edit mariadb@10.10`

Remove the line:

disable! date: "2024-07-26", because: :...

And install it with:

`HOMEBREW_NO_INSTALL_FROM_API=1 brew install mariadb@10.10`

**Comment by 1rocketdude** [comment]
I concur. Uninstalling the current version and installing ***@***.*** worked for me. I have very large databases; happy that it worked.

**Comment by EmkeDevani** [comment]
Is there a way to get a backup of my databases? Mariadb stopped working so i can't export sql files. I have a backup of my mysql folder but as soon as i put this folder back, Mariadb crashes again.

**Reply by reckart** [reply]
```
[mariadb]
innodb_force_recovery = 6
```

and then dump the DB and restore it into a new installation. Mind, there will be some loss. I lost foreign key declarations, others lost other things.

**Comment by benzdouglas** [comment]
Just want to say thank you to everyone in this thread, I have wasted so much of my life on this issue over the past few weeks. Copilot is useless by the way at troubleshooting these kind of issues and I've learned my lesson.

**Comment by milhouse1337** [comment]
MariaDB `10.11.9` is out, the package for brew should be there soon.
https://mariadb.com/kb/en/mariadb-10-11-9-changelog/

**Comment by dennis-berg** [comment]
Note that mariadb 10.6 does not yet support the UUID data type. Was a problem for me so I turned to the next version supported by homebrew that supports UUID, which was 10.11 as of today. Followed the instructions from [mazzelman](https://github.com/mazzelman) above. Thanks.

update: 10.11 failed, had to rerun on the 10.6 (and opt for changing the uuid datatypes locally).

update 2: so now i have the workaround which look like this:
to turn mariadb on: 
`brew services start mariadb@10.6`
to turn mariadb off: 
`brew services stop mariadb@10.6`
`ps aux | grep mariadb`
find the process id with mariadbd
`kill -9 [process id]`

**Comment by 1rocketdude** [comment]
are you saying that mariadb 11.1 does not have the data corruption on shutdown behavior?

**Comment by damient** [comment]
MariaDB released new versions on 2024-08-15 that correct the issue. Is it possible to have it with Homebrew Formulae ?

**Comment by 1rocketdude** [comment]
I installed using: brew install ***@***.***

the shutdown error is corrected in this version.

Beware that when first starting ***@***.***, it will fail if ***@***.*** was not shutdown properly and the database log file flushed. So, warning to dump all databases from ***@***.*** to a file before upgrading.

**Comment by mpreitz** [comment]
Removed old installation, deleted mysql data folder.
Installed mariadb:
brew install mariadb

Imported one database, afterwards stop  service and checked with ps: no more mariadb instances running.
Startet mariadb service:

> 2024-08-20 15:05:10 0 [ERROR] InnoDB: Missing FILE_CHECKPOINT(1951515931) at 1951515931
2024-08-20 15:05:10 0 [ERROR] InnoDB: Log scan aborted at LSN 1951515931
2024-08-20 15:05:10 0 [ERROR] InnoDB: Plugin initialization aborted with error Generic error
2024-08-20 15:05:10 0 [Note] InnoDB: Starting shutdown...
2024-08-20 15:05:10 0 [ERROR] Plugin 'InnoDB' registration as a STORAGE ENGINE failed.

So, the process is stopped but nevertheless something is wrong and the logfile gets corrupted.

Still broken and still not usable.

**Reply by reckart** [reply]
@mpreitz Please verify which version of MariaDB you are now running. If it is one that should have the issue fixed, open a new issue in the MariaDB tracker and tell them how to reproduce the problem. Link to the old (closed) issue in the new one and tell them it still does not work. Maybe check if somebody else has already opened a new issue before you create one. If you found an existing issue or opened a new one, please link it here.

**Reply by reckart** [reply]
I would say if you just did a `brew install mariadb`, there is a good chance you got an old version still, namely [`11.4.2`](https://formulae.brew.sh/formula/mariadb#default) - and the fix 

![Screenshot 2024-08-20 at 21 47 12](https://github.com/user-attachments/assets/e192f098-faf0-4536-9eab-e818f3cb27e3)

However, the fix is in [10.11.9](https://jira.mariadb.org/issues/?jql=project+%3D+MDEV+AND+fixVersion+%3D+10.11.9), [11.1.6](https://jira.mariadb.org/issues/?jql=project+%3D+MDEV+AND+fixVersion+%3D+11.1.6), [11.2.5](https://jira.mariadb.org/issues/?jql=project+%3D+MDEV+AND+fixVersion+%3D+11.2.5), **[11.4.3](https://jira.mariadb.org/issues/?jql=project+%3D+MDEV+AND+fixVersion+%3D+11.4.3)**, [11.5.2](https://jira.mariadb.org/issues/?jql=project+%3D+MDEV+AND+fixVersion+%3D+11.5.2), [11.6.1](https://jira.mariadb.org/issues/?jql=project+%3D+MDEV+AND+fixVersion+%3D+11.6.1) according to https://jira.mariadb.org/browse/MDEV-34422

**Reply by mpreitz** [reply]
Sorry, forgot to post the version. It was 11.4.2 (which was installed per default when not specifing a version). Now the version was bumped to 11.5.2. Tried again (with a complete removal, incl. data folder before) and now MariaDB stops and starts flawless without corrupting the innodb logs.

**Comment by 1rocketdude** [comment]
very instructive - thank you.

so…wait until 11.4.3 becomes part of ‘brew update’, then things should be fixed?

I upgraded from mariadb10.6 to mariadb10.11 on one machine and all is well….I would rather get back to 11.4 when it’s functional.

**Reply by troyxmccall** [reply]
@1rocketdude you have a browser extension or MIM proxy that's wiping out the formula and version numbers in your posts


ie: 
> ` I upgraded from ***@***.*** to ***@***.*** on one machine and all is well….I would rather get back to 11.4 when it’s functional.`

**Reply by 1rocketdude** [reply]
weird. the internet is complex. '@' signs must trigger a privacy shield.

**Comment by 1rocketdude** [comment]
anyone try mariadb@11.5, which appears to have been released recently?
is the shutdown bug gone from this release?
I see that 11.4.3 is now available as well.

**Comment by renekreijveld** [comment]
I have tested version 11.5.2 and that seems to run fine.
I was running 10.6.19.
I made a mysqldump of all my databases and then uninstalled 10.6.19 with ```brew uninstall mariadb@10.6```
I then deleted these files and folders:

```
rm /opt/homebrew/etc/my.cnf
rm -rf /opt/homebrew/var/mysql
rm -rf /opt/homebrew/etc/my.cnf.d
```

I then installed 11.5.2: ```brew install mariadb```
I did the following tests:
```brew services stop mariadb```
There were no more hanging mariadb or mysql processes.
After starting mariadb no tables were corrupt.
I then rebooted my Mac. After reboot mariadb was running again and again had no corrupt tables.

**Reply by 1rocketdude** [reply]
I concur. I upgraded to 11.5.2 and no table corruption.

**Reply by hestroy** [reply]
Me too.

**Comment by ksn135** [comment]
Is there any possibility to switch from version 10.6 to the latest version without erasing the folder with databases?

**Reply by 1rocketdude** [reply]
I have never seen a recommendation for or against upgrading without a mysqldump and reload. In practice, it has worked for me. I just upgraded from 10.11 to 11.5.2 by simply stopping the mariadb@10.11 server, uninstalling, installing version 11.5.2 and starting. My databases are intact.
It's certainly worth the time to dump all databases just in case.

**Reply by reckart** [reply]
You may want to run [mariadb-upgrade](https://mariadb.com/kb/en/mariadb-upgrade/) after an upgrade though - not sure if the brew formula does that automatically.

**Reply by renekreijveld** [reply]
I would always make a backup first. There are plenty scripts te be found online. Here's what MariaDB suggests:
https://mariadb.com/kb/en/making-backups-with-mariadb-dump/

**Reply by ksn135** [reply]
Thank you, everyone. Updated to the latest version (11.5.2).
No data corruption after restart.
